src/dialect/mssql.rs

```diff
@@ -215,6 +225,59 @@ impl MsSqlDialect {
         }))
     }
 
+    /// Parse `CREATE TRIGGER` for [MsSql]
+    ///
+    /// [MsSql]: https://learn.microsoft.com/en-us/sql/t-sql/statements/create-trigger-transact-sql
+    fn parse_create_trigger(
+        &self,
+        parser: &mut Parser,
+        or_alter: bool,
+    ) -> Result<Statement, ParserError> {
+        let name = parser.parse_object_name(false)?;
+        parser.expect_keyword_is(Keyword::ON)?;
+        let table_name = parser.parse_object_name(false)?;
+        let period = parser.parse_trigger_period()?;
+        let events = parser.parse_comma_separated(Parser::parse_trigger_event)?;
+
+        parser.expect_keyword_is(Keyword::AS)?;
+
+        let trigger_statements_body = if parser.peek_keyword(Keyword::BEGIN) {
```

Comment:
Similar approach in the other PR for stored procedures that may/not have begin/end tokens https://github.com/apache/datafusion-sqlparser-rs/pull/1834/files#r2069458988

Reply:
I also updated the related stored procedure PR accordingly: https://github.com/apache/datafusion-sqlparser-rs/pull/1834#issuecomment-2847567460